File: src/com/mojang/escape/level/block/WaterBlock.java
package com.mojang.escape.level.block;

import com.mojang.escape.Art;
import com.mojang.escape.entities.*;

public class WaterBlock extends Block {
	int steps = 0;

	public WaterBlock() {
		setBlocksMotion(true);
	}

	public void tick() {
		super.tick();
		steps--;
		if (steps <= 0) {
			setFloorTex(8 + getRandom().nextInt(3));
			setFloorCol(Art.INSTANCE.getCol(0x0000ff));
			steps = 16;
		}
	}

	public boolean blocks(Entity entity) {
		if (entity instanceof Player) {
			if (((Player) entity).getSelectedItem() == Item.Flippers) return false;
		}
		if (entity instanceof Bullet) return false;
		return getBlocksMotion();
	}

	public double getFloorHeight(Entity e) {
		return -0.5;
	}

	public double getWalkSpeed(Player player) {
		return 0.4;
	}

}

File: src/com/mojang/escape/level/block/WaterBlock.kt
package com.mojang.escape.level.block

import com.mojang.escape.Art
import com.mojang.escape.entities.Bullet
import com.mojang.escape.entities.Entity
import com.mojang.escape.entities.Item
import com.mojang.escape.entities.Player

class WaterBlock: Block() {
    private var steps = 0

    init {
        blocksMotion = true
    }

    override fun tick() {
        super.tick()
        steps--
        if (steps <= 0) {
            floorTex = 8 + random.nextInt(3)
            floorCol = Art.getCol(0x0000FF)
            steps = 16
        }
    }

    override fun blocks(entity: Entity): Boolean {
        if (entity is Player) {
            if (entity.selectedItem == Item.Flippers) {
                return false
            }
        }
        if (entity is Bullet) {
            return false
        }
        return blocksMotion
    }

    override fun getFloorHeight(e: Entity): Double {
        return -0.5
    }

    override fun getWalkSpeed(player: Player): Double {
        return 0.4
    }
}